pkg/api/router.go

```diff
@@ -37,21 +37,35 @@ func (s *server) setupRouting() {
 	})
 
 	handle(router, "/files", jsonhttp.MethodHandler{
-		"POST": http.HandlerFunc(s.fileUploadHandler),
+		"POST": web.ChainHandlers(
+			s.instrumentOpenTracing(),
+			http.HandlerFunc(s.fileUploadHandler),
```

Comment:
This should be `web.FinalHandlerFunc(s.fileUploadHandler)`, like in other places below where compiler complains.